shards/shards.go

```diff
@@ -522,28 +523,38 @@ func (ss *shardedSearcher) StreamSearch(ctx context.Context, q query.Q, opts *zo
 	}
 	defer proc.Release()
 	tr.LazyPrintf("acquired process")
+
+	shards := ss.getShards()
+	if len(shards) == 0 {
+		return nil
+	}
```

Comment:
I think we still wanna send the Wait time stat event here. Maybe we just need to set MaxPendingPriority to 0 in this case and continue with the rest of the code path.

Reply:
Done. I set `MaxPendingPriority=math.Inf(-1)` instead of 0 to be consistent with `pending.max()`.